c10/core/ScalarType.h

```diff
@@ -77,7 +78,14 @@ namespace c10 {
   _(c10::Float8_e4m3fnuz, Float8_e4m3fnuz) /* 26 */      \
   _(uint16_t, UInt16) /* 27 */                           \
   _(uint32_t, UInt32) /* 28 */                           \
-  _(uint64_t, UInt64) /* 29 */
+  _(uint64_t, UInt64) /* 29 */                           \
+  _(c10::uint1_t, UInt1) /* 30 */                        \
```

Comment:
I would prefer these to have dummy in the name to indicate that they don't actually work. It should also be feasible to define only one struct templates over int